notifier.rst

```diff
@@ -0,0 +1,564 @@
+
+                    <framework:channel-policy>
+                        <!-- Use SMS, Slack and Email for urgent notifications -->
+                        <framework:urgent>sms</framework:urgent>
+                        <framework:urgent>chat/slack</framework:urgent>
+                        <framework:urgent>email</framework:urgent>
+
+                        <!-- Use Slack for highly important notifications -->
+                        <framework:high>chat/slack</framework:high>
+
+                        <!-- Use browser for medium and low notifications -->
+                        <framework:medium>browser</framework:medium>
+                        <framework:low>browser</framework:low>
+                    </framework:channel-policy>
+                </framework:notifier>
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        # config/packages/notifier.php
+        $container->loadFromExtension('framework', [
+            'notifier' => [
+                // ...
+                'channel_policy' => [
+                    // Use SMS, Slack and email for urgent notifications
+                    'urgent' => ['sms', 'chat/slack', 'email'],
+
+                    // Use Slack for highly important notifications
+                    'high' => ['chat/slack'],
+
+                    // Use browser for medium and low notifications
+                    'medium' => ['browser'],
+                    'low' => ['browser'],
+                ],
+            ],
+        ]);
+
+Now, whenever the notification's importance is set to "high", it will be
+sent using the Slack transport::
+
+    // ...
+    class InvoiceController extends AbstractController
+    {
+        /**
+         * @Route("/invoice/create")
+         */
+        public function invoice(NotifierInterface $notifier)
+        {
+            // ...
+
+            $notification = (new Notification('New Invoice'))
+                ->content('You got a new invoice for 15 EUR.')
+                ->importance(Notification::IMPORTANCE_HIGH);
+
+            $notifier->send($notification, new Recipient('wouter@wouterj.nl'));
+
+            // ...
+        }
+    }
+
+Customize Notifications
+-----------------------
+
+You can extend the ``Notification`` or ``Recipient`` base classes to
+customize their behavior. For instance, you can overwrite the
+``getChannels()`` method to only return ``sms`` if the invoice price is
+very high and the recipient has a phonenumber::
+
+    namespace App\Notifier;
+
+    use Symfony\Component\Notifier\Notification\Notification;
+    use Symfony\Component\Notifier\Recipient\Recipient;
+
+    class InvoiceNotification extends Notification
+    {
+        private $price;
+
+        public function __construct(int $price)
+        {
+            $this->price = $price;
+        }
+
+        public function getChannels(Recipient $recipient)
+        {
+            if ($this->price > 10000 && $receiver->hasPhone()) {
```

Comment:
I don't see that `Recipient` would have `hasPhone` method :thinking:

Reply:
Thanks! Fixed the code :)